test/saithrift/tests/saiacl.py

```diff
@@ -15,114 +15,427 @@
+                                                   mac_src = rewrite_mac1)
+        print "Source MAC address dropped successfully"
+
+@group('acl')
+class testAclSrcMacForward(sai_base_test.ThriftInterfaceDataPlane, aclTestBase):
+    def runTestFunc(self):
+        super(testAclSrcMacForward, self).runTestFunc(stage = SAI_ACL_STAGE_INGRESS,
+                                                      action = SAI_PACKET_ACTION_FORWARD,
+                                                      in_ports = [port_list[PORT_OUT], port_list[PORT_IN]],
+                                                      mac_src = rewrite_mac1)
+        print "Source MAC address forwarded successfully"
+
+@group('acl')
+class testAclDstMacDrop(sai_base_test.ThriftInterfaceDataPlane, aclTestBase):
+    def runTestFunc(self):
+        super(testAclDstMacDrop, self).runTestFunc(stage = SAI_ACL_STAGE_INGRESS,
+                                                   action = SAI_PACKET_ACTION_DROP,
+                                                   in_ports = [port_list[PORT_OUT], port_list[PORT_IN]],
+                                                   mac_dst = router_mac)
+        print "Destination MAC address dropped successfully"
+
+@group('acl')
+class testAclDstMacForward(sai_base_test.ThriftInterfaceDataPlane, aclTestBase):
+    def runTestFunc(self):
+        super(testAclDstMacForward, self).runTestFunc(stage = SAI_ACL_STAGE_INGRESS,
+                                                      action = SAI_PACKET_ACTION_FORWARD,
+                                                      in_ports = [port_list[PORT_OUT], port_list[PORT_IN]],
+                                                      mac_dst = router_mac)
+        print "Destination MAC address forwarded successfully"
+
+
+###############################################################################################################
+################################# testing COPY action (SAI_ACL_STAGE_INGRESS) #################################
+###############################################################################################################
 
-            # cleanup
-            sai_thrift_remove_route(self.client, vr_id, addr_family, ip_addr1, ip_mask1, nhop1)
-            self.client.sai_thrift_remove_next_hop(nhop1)
-            sai_thrift_remove_neighbor(self.client, addr_family, rif_id1, ip_addr1, dmac1)
+@group('acl')
+class testAclSrcIpCopy(sai_base_test.ThriftInterfaceDataPlane, aclTestBase):
+    def runTestFunc(self):
+        super(testAclSrcIpCopy, self).runTestFunc(stage = SAI_ACL_STAGE_INGRESS,
+                                                  action = SAI_PACKET_ACTION_COPY,
+                                                  in_ports = [port_list[PORT_OUT], port_list[PORT_IN]],
+                                                  ip_src = IP_SRC,
+                                                  ip_src_mask = IP_MASK)
+        print "Source IPv4 address copied to CPU"
```

Comment:
>copied to CPU [](start = 35, length = 13)

we did not really test the packet has been copied to cpu, but that's fine. we can improve this later.